event.go

```diff
@@ -60,9 +62,9 @@ type EventBuilder struct {
 	// The state_key of the event if the event is a state event or nil if the event is not a state event.
 	StateKey *string `json:"state_key,omitempty"`
 	// The events that immediately preceded this event in the room history.
-	PrevEvents []EventReference `json:"prev_events"`
+	PrevEvents interface{} `json:"prev_events"`
 	// The events needed to authenticate this event.
-	AuthEvents []EventReference `json:"auth_events"`
+	AuthEvents interface{} `json:"auth_events"`
```

Comment:
I really would prefer we didn't use `interface{}` and lose all types. Why are we doing this, there's zero justifcation or docs here and I would expect to see it.

Reply:
I'll add comments here - the main reason is that `prev_events` and `auth_events` are different formats in v1/v2 vs. v3 onwards. In v3 onwards, they are no longer `[]EventReference`, but `[]string`.